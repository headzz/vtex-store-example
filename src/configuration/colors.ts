const BLACK = '#000000'
const DARK_GREY = '#5B5B5B'
const GREEN_JUSSI = '#03FFA5'
const GREY_LVL1 = '#F2F2F2'
const PINK_JUSSI = '#EE0E7D'
const TRANSPARENT = 'transparent'
const WHITE = '#ffffff'

export default {
  BLACK,
  DARK_GREY,
  GREEN_JUSSI,
  GREY_LVL1,
  PINK_JUSSI,
  WHITE,
  TRANSPARENT
}
