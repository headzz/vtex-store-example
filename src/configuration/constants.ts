export const BLACK = '#000000'
export const WHITE = '#ffffff'
export const GREY_LVL1 = '#F2F2F2'
export const GREEN_JUSSI = '#03FFA5'
export const DARK_GREY = '#5B5B5B'
export const PINK_JUSSI = '#EE0E7D'
