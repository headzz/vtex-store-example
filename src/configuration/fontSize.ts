const SIZE_TEXT_SMALL = '1.2rem'
const SIZE_TEXT_NORMAL = '1.6rem'

const SIZE_TITLE_SMALL = '3.2rem'
const SIZE_TITLE_NORMAL = '6.4rem'

export default {
  SIZE_TEXT_SMALL,
  SIZE_TEXT_NORMAL,
  SIZE_TITLE_NORMAL,
  SIZE_TITLE_SMALL
}
