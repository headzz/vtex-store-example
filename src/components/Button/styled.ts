import { colors } from 'configuration'
import styled from 'styled-components'

type Link = {
  round: boolean
  icon: 'search' | 'cart' | ''
}
export const ButtonStyled = styled('button')<Link>`
  border: 0;
  border-radius: ${({ round }) => (round ? '50%' : '0')};
  background-color: ${colors.TRANSPARENT};
  outline: 0;
  padding: ${({ icon }) => (icon !== '' ? '' : '1.2rem 2rem')};
  width: ${({ icon }) => (icon !== '' ? 'fit-content' : '100%')};
`
