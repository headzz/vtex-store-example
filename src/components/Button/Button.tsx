import Cart from 'components/Icons/Cart'
import Search from 'components/Icons/Search'
import Text from 'components/Text'
import { ButtonStyled } from './styled'

const availableIcons = {
  search: <Search />,
  cart: <Cart />
}

type Props = {
  color: string
  icon: 'search' | 'cart' | ''
  round: boolean
  clickHandler: (event: React.MouseEvent<HTMLButtonElement>) => void
}
/**
 * @component
 * @name Button
 * @description Defines button component.
 * @namespace components components/Button
 * @param {element} children Defining the text from the button.
 * @param {string} color Sets the color of the text inside button.
 * @param {string} icon Sets an icon to the button.
 * @param {boolean} round Sets rounded button.
 * @param {React.MouseEvent<HTMLButtonElement>} clickHandler Creates on click action.
 *
 */
export const Button: React.FC<Props> = ({
  children,
  clickHandler,
  color,
  icon,
  round
}) => (
  <>
    {!icon && (
      <ButtonStyled round={round} onClick={clickHandler}>
        <Text color={color} title="" text="">
          {children}
        </Text>
      </ButtonStyled>
    )}

    {icon !== '' && (
      <ButtonStyled round={round} onClick={clickHandler}>
        {availableIcons[icon]}
      </ButtonStyled>
    )}
  </>
)
