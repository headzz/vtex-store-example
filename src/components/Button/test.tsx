import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { colors } from 'configuration'
import Button from './Button'
describe('Button', () => {
  it('should render component button icon', () => {
    const handleClick = jest.fn()
    render(
      <Button
        color={colors.BLACK}
        clickHandler={handleClick}
        round={false}
        icon=""
      >
        Teste
      </Button>
    )

    expect(screen.getByRole('button', { name: /teste/i })).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /teste/i })).toMatchSnapshot()
  })

  it('should click the button and execute function once', () => {
    const handleClick = jest.fn()
    render(
      <Button
        color={colors.BLACK}
        clickHandler={handleClick}
        round={false}
        icon=""
      >
        Teste
      </Button>
    )

    userEvent.click(screen.getByRole('button', { name: /teste/i }))

    expect(handleClick).toHaveBeenCalledTimes(1)
    expect(screen.getByRole('button', { name: /teste/i })).toMatchSnapshot()
  })

  it('should render button with icon', () => {
    const handleClick = jest.fn()
    render(
      <Button
        color={colors.BLACK}
        clickHandler={handleClick}
        round={true}
        icon="search"
      >
        Teste
      </Button>
    )

    expect(screen.getByAltText('Ã­cone de pesquisa')).toBeInTheDocument()
    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(screen.getByRole('button')).toMatchSnapshot()
  })
})
