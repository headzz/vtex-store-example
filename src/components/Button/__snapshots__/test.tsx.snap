// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should click the button and execute function once 1`] = `
.c1 {
  color: #000000;
  font-size: 1.6rem;
  font-weight: 500;
}

.c0 {
  border: 0;
  border-radius: 0;
  background-color: transparent;
  outline: 0;
  padding: 1.2rem 2rem;
  width: 100%;
}

<button
  class="c0"
>
  <p
    class="c1"
    color="#000000"
    title=""
  >
    Teste
  </p>
</button>
`

exports[`Button should render button with icon 1`] = `
.c0 {
  border: 0;
  border-radius: 50%;
  background-color: transparent;
  outline: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

<button
  class="c0"
>
  <div
    style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
  >
    <div
      style="box-sizing: border-box; display: block; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        role="presentation"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
        style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
      />
    </div>
    <img
      alt="Ã­cone de pesquisa"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
  </div>
</button>
`

exports[`Button should render component button icon 1`] = `
.c1 {
  color: #000000;
  font-size: 1.6rem;
  font-weight: 500;
}

.c0 {
  border: 0;
  border-radius: 0;
  background-color: transparent;
  outline: 0;
  padding: 1.2rem 2rem;
  width: 100%;
}

<button
  class="c0"
>
  <p
    class="c1"
    color="#000000"
    title=""
  >
    Teste
  </p>
</button>
`
